src/main/java/sorald/processor/ThreadLocalWithInitial.java

```diff
@@ -0,0 +1,81 @@
+package sorald.processor;
+
+import java.util.List;
+import java.util.function.Supplier;
+import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.code.CtLambda;
+import spoon.reflect.code.CtNewClass;
+import spoon.reflect.code.CtReturn;
+import spoon.reflect.code.CtStatement;
+import spoon.reflect.code.CtTypeAccess;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.reference.CtTypeReference;
+
+@ProcessorAnnotation(key = "S4065", description = "\"ThreadLocal.withInitial\" should be preferred")
+public class ThreadLocalWithInitial extends SoraldAbstractProcessor<CtNewClass<?>> {
+
+    @Override
+    protected void repairInternal(CtNewClass<?> newClass) {
+        CtClass<?> innerClass = newClass.getAnonymousClass();
+        if (hasNoFields(innerClass) && hasOnlyConstructorAndSingleMethod(innerClass)) {
```

Comment:
Apparently, we don't need these conditions as well. As @khaes-kth suggested, I checked if SonarLint reports a violation if there are more than 2 type members. It does not. For example, the following snippet does not report any sonar violation.

```java
ThreadLocal<List<String>> myThreadLocal = new ThreadLocal<>() {
            private static final String X = "1";
            @Override
            protected List<String> initialValue() {
                System.setProperty(X, "abc");
                return new ArrayList<>();
            }
        };
```